### PR TITLE
Backport zen 2.0.22 to zendoo

### PR DESCRIPTION
**Pull request by abi87**
PR name should say it all

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -818,15 +821,20 @@ bool AreInputsStandard(const CTransactionBase& txBase, const CCoinsViewCache& ma
     if (txBase.IsCoinBase())
         return true; // Coinbases don't use vin normally
 
-    for(const CTxIn& in : txBase.GetVin()) {
-        const CTxOut& prev = mapInputs.GetOutputFor(in);
+    for (unsigned int i = 0; i < txBase.GetVin().size(); i++)
```
Comment (alsala) [inline src/main.cpp:824]: Consider using:
 const CTxin& in = txBase.GetVin().at(i)
for better reading and also since it is used twice in this block

**Review by alsala — CHANGES_REQUESTED** [review]

File: src/miner.cpp [hunk 2]
```diff
@@ -864,7 +864,7 @@ void static BitcoinMiner()
 {
     LogPrintf("HorizenMiner started\n");
     SetThreadPriority(THREAD_PRIORITY_LOWEST);
-    RenameThread("zen-miner");
+    RenameThread("horizzen-miner");
```
[the remark this answers is not in the document]
Comment (abi87) [reply src/miner.cpp]: Done

File: src/script/standard.cpp [hunk 3]
```diff
@@ -162,73 +175,198 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, vector<vector<unsi
...
+                    }
+                    catch(const scriptnum_error& e)
+                    {
+                        LogPrintf("%s: %s():%d - OP_CHECKBLOCKATHEIGHT nHeight 0x%s not minimally encoded (err=%s)\n",
+                            __FILE__, __func__, __LINE__, HexStr(vchBlockHeight.begin(), vchBlockHeight.end()), e.what());
+                        break;
+                    }
+                    catch(...)
+                    {
+                        LogPrint("%s: %s():%d - unexpected exception\n", __FILE__, __func__, __LINE__);
```
[the remark this answers is not in the document]
Comment (abi87) [reply src/script/standard.cpp]: Done

File: src/main.cpp (hunk 1 above)
Comment (abi87) [reply src/main.cpp:824]: Done

File: doc/release-notes/release-notes-2.0.22.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      Update src/init.cpp
+
+ca333 (1):
+      update libsodium to v1.0.18
+
+cronicc (19):
+      Update openssl from 1.1.1d to 1.1.1g
+      Update univalue to v1.1.1
+      Fix MacOS build issue "conversion from 'size_t' (aka 'unsigned long') to 'const UniValue' is ambiguous"
+      Replace calls to deprecated method push_back(Pair()) with pushKV()
```
[the remark this answers is not in the document]
Comment (abi87) [reply doc/release-notes/release-notes-2.0.22.md:88]: I would rather postpone this and let infra guys update the release notes as soon as we release the whole thing

**Review by alsala — APPROVED** [review]

**Pull request closed without merging**